packages/cubit/lib/src/cubit.dart

```diff
@@ -25,11 +27,34 @@ abstract class Cubit<T> extends Stream<T> {
 
   T _state;
 
+  /// Called whenever a [transition] occurs with the given [transition].
+  /// A [transition] occurs when a new `state` is emitted.
+  /// [onTransition] is called before the `state` of the `cubit` is updated.
+  /// A great spot to add logging/analytics at the individual `cubit` level.
+  ///
+  /// **Note: `super.onTransition` should always be called last.**
+  /// ```dart
+  /// @override
+  /// void onTransition(Transition transition) {
+  ///   // Custom onTransition logic goes here
+  ///
+  ///   // Always call super.onTransition with the current transition
+  ///   super.onTransition(transition);
+  /// }
+  /// ```
+  @mustCallSuper
+  void onTransition(Transition<Null, T> transition) {}
+
   /// Updates the [state] of the `cubit` to the provided [state].
   /// [emit] does nothing if the `cubit` has been closed.
   @protected
   void emit(T state) async {
     if (state == _state || _controller.isClosed) return;
+    onTransition(Transition(
+      currentState: _state,
+      event: null,
```

Comment:
What's the thinking behind the event parameter? Cubit doesn't have a concept up events, right? Is this somehow meant to support the needs of Bloc?

Reply:
We might be able to determine the name of the method from which the state was emitted so I was thinking it made sense to leave the door open for that possibility 😛

Reply:
Gotchaaaaaa. Cool. Thanks for sharing.